src/components/Sidebar.tsx

```diff
@@ -7,7 +7,79 @@ import { IRecipe } from "../interfaces/Recipe";
 import BackBtn from "./BackBtn";
 import BackgroundTexture from "../assets/bgTexture.svg";
 
-const SidebarContainer = styled.div`
+interface IconProps {
+  firstClick: boolean;
+  sideBarState: boolean;
+}
+interface SidebarProps {
+  selectedIngredients: string[];
+  setSelectedIngredients: Function;
+  searchByIngredients: Function;
+  setSearchedRecipes: Function;
+  setUserSearchedRecipes: Function;
+  selectedRecipes: IRecipe[];
+  getRecipeDetails: Function;
+  userSearchedRecipes: Boolean;
+  renderIngredientBlobs: Function;
+  setHideRecipeDetails: Function;
+  setSelectedRecipes: Function;
+  removeIngredient: Function;
+}
+
+interface SidebarContainerProps {
+  firstClick: boolean;
+  sideBarState: boolean;
+}
+
+const sideBarIn = keyframes`
+  from {transform: translateY(100%);}
+  to {transform: translateY(0%)}
+`
+
+const sideBarOut = keyframes`
+  from {transform: translateY(0%)}
+  to {transform: translateY(100%)}
+`
+
+const upperIn = keyframes`
+from {transform: translateY(0%)}
+to {transform: translateY(400%)}
+`;
```

Comment:
small fix, you forgot to indent on line 45, 46, just thought I'd let you know :)